spec.emu

```diff
@@ -112,24 +136,40 @@ contributors: Chengzhong Wu, Hemanth HM, Ecma International
         <h1>The AggregateError Constructor</h1>
 
         <emu-clause id="sec-aggregate-error">
-          <h1>AggregateError ( _errors_, _message_<ins> [ , _cause_ ]</ins> )</h1>
-          <p>When the *AggregateError* function is called with arguments _errors_<del> and</del><ins>,</ins> _message_<ins>, and _cause_</ins>, the following steps are taken:</p>
+          <h1>AggregateError ( _errors_, _message_<ins> [ , _options_ ]</ins> )</h1>
+          <p>When the *AggregateError* function is called with arguments _errors_<del> and</del><ins>,</ins> _message_<ins>, and optional argument _options_</ins>, the following steps are taken:</p>
           <emu-alg>
             1. If NewTarget is *undefined*, let _newTarget_ be the active function object; else let _newTarget_ be NewTarget.
             1. Let _O_ be ? OrdinaryCreateFromConstructor(_newTarget_, *"%AggregateError.prototype%"*, &laquo; [[ErrorData]] &raquo;).
             1. If _message_ is not *undefined*, then
               1. Let _msg_ be ? ToString(_message_).
-              1. Let _msgDesc_ be the PropertyDescriptor { [[Value]]: _msg_, [[Writable]]: *true*, [[Enumerable]]: *false*, [[Configurable]]: *true* }.
-              1. Perform ! DefinePropertyOrThrow(_O_, *"message"*, _msgDesc_).
-            1. <ins>If _cause_ is not *undefined*, then</ins>
-              1. <ins>Let _causeDesc_ be the PropertyDescriptor { [[Value]]: _cause_, [[Writable]]: *true*, [[Enumerable]]: *false*, [[Configurable]]: *true* }.</ins>
-              1. <ins>Perform ! DefinePropertyOrThrow(_O_, `"cause"`, _causeDesc_).</ins>
+              1. <del>Let _msgDesc_ be the PropertyDescriptor { [[Value]]: _msg_, [[Writable]]: *true*, [[Enumerable]]: *false*, [[Configurable]]: *true* }.</del>
+              1. <del>Perform ! DefinePropertyOrThrow(_O_, *"message"*, _msgDesc_).</del>
+              1. <ins>Perform ! CreateNonEnumerableDataProperty(_O_, *"message"*, _msg_).</ins>
+            1. <ins>Perform ? ExtractErrorCause(_O_, _options_). </ins>
             1. Let _errorsList_ be ? IterableToList(_errors_).
             1. Perform ! DefinePropertyOrThrow(_O_, *"errors"*, PropertyDescriptor { [[Configurable]]: *true*, [[Enumerable]]: *false*, [[Writable]]: *true*, [[Value]]: ! CreateArrayFromList(_errorsList_) }).
             1. Return _O_.
           </emu-alg>
         </emu-clause>
       </emu-clause>
     </emu-clause>
+
+    <emu-clause id="sec-errorobjects-abstracts">
+      <h1>Abstract Operations for Error Objects</h1>
+
+
+      <emu-clause id="sec-errorobjects-extract-error-cause">
+        <h1><ins>ExtractErrorCause ( _O_, _options_ )</ins></h1>
```

Comment:
"Extract" implies to me that this is going to return a value, not have a side effect. "Add" or "Install" would be better, I think.

Reply:
Renamed to `InstallErrorCause`.